docs/client/overview.rst

```diff
@@ -0,0 +1,622 @@
+
+    >>> frontiers = project.frontiers
+
+Get all frontiers from a project to iterate through it::
+
+    >>> frontiers.iter()
+    <list_iterator at 0x103c93630>
+
+List all frontiers::
+
+    >>> frontiers.list()
+    ['test', 'test1', 'test2']
+
+Get a frontier by name::
+
+    >>> frontier = frontiers.get('test')
+    >>> frontier
+    <scrapinghub.client.Frontier at 0x1048ae4a8>
+
+Get an iterator to iterate through a frontier slots::
+
+    >>> frontier.iter()
+    <list_iterator at 0x1030736d8>
+
+List all slots::
+
+    >>> frontier.list()
+    ['example.com', 'example.com2']
+
+Get a frontier slot by name::
+
+    >>> slot = frontier.get('example.com')
+    >>> slot
+    <scrapinghub.client.FrontierSlot at 0x1049d8978>
+
+Add a request to the slot::
+
+    >>> slot.queue.add([{'fp': '/some/path.html'}])
+    >>> slot.flush()
+    >>> slot.newcount
+    1
+
+``newcount`` is defined per slot, but also available per frontier and globally::
+
+    >>> frontier.newcount
+    1
+    >>> frontiers.newcount
+    3
+
+Add a fingerprint only to the slot::
+
+    >>> slot.fingerprints.add(['fp1', 'fp2'])
+    >>> slot.flush()
+
+There are convenient shortcuts: ``f`` for ``fingerprints`` and ``q`` for ``queue``.
+
+Add requests with additional parameters::
+
+    >>> slot.q.add([{'fp': '/'}, {'fp': 'page1.html', 'p': 1, 'qdata': {'depth': 1}}])
+    >>> slot.flush()
+
+To retrieve all requests for a given slot::
+
+    >>> reqs = slot.q.iter()
+
+To retrieve all fingerprints for a given slot::
+
+    >>> fps = slot.f.iter()
+
+To list all the requests use ``list()`` method (similar for ``fingerprints``)::
+
+    >>> fps = slot.q.list()
+
+To delete a batch of requests::
+
+    >>> slot.q.delete('00013967d8af7b0001')
+
+To delete the whole slot from the frontier::
+
+    >>> slot.delete()
+
+Flush data of the given frontier::
+
+    >>> frontier.flush()
+
+Flush data of all frontiers of a project::
+
+    >>> frontiers.flush()
+
+Close batch writers of all frontiers of a project::
+
+    >>> frontiers.close()
+
+Frontiers are available on project level only.
+
+.. _job-tags:
+
+Tags
+----
+
+Tags is a convenient way to mark specific jobs (for better search, postprocessing etc).
+
+To mark a job with tag ``consumed``::
+
+    >>> job.update_tags(add=['consumed'])
+
+To mark all spider jobs with tag ``consumed``::
+
+    >>> spider.jobs.update_tags(add=['consumed'])
+
+To remove existing tag ``existing`` for all spider jobs::
+
+    >>> spider.jobs.update_tags(remove=['existing'])
+
+Modifying tags is available on spider/job levels.
+
+
+Exceptions
+----------
+
+scrapinghub.exceptions.ScrapinghubAPIError
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Base exception class for all other exceptions listed below.
+
+
+scrapinghub.exceptions.BadRequest
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Usually raised in case of 400 response from API.
+
+
+scrapinghub.exceptions.Unauthorized
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Not enough access to some resources.
```

Comment:
Confusing, better reword it like `Request lacks valid authentication credentials for the target resource`